samples/enable-autoupgrade-changes/createUIDefinition.json

```diff
@@ -0,0 +1,234 @@
+							{
+								"name": "enableAutoScaling",
+								"type": "Microsoft.Common.CheckBox",
+								"label": "Enable auto scaling",
+								"toolTip": "Enable auto scaling",
+								"defaultValue": true
+							},
+							{
+								"name": "vmCount",
+								"type": "Microsoft.Common.Slider",
+								"min": 1,
+								"max": 10,
+								"label": "VMCount",
+								"subLabel": "",
+								"defaultValue": 1,
+								"showStepMarkers": false,
+								"toolTip": "Specify VM count",
+								"constraints": {
+									"required": false
+								},
+								"visible": true
+							}
+						],
+						"visible": "[basics('createNewCluster')]"
+					}
+				]
+			},
+			{
+				"name": "applicationDetails",
+				"label": "Cluster Application Details",
+				"elements": [
+					{
+						"name": "extensionResourceName",
+						"type": "Microsoft.Common.TextBox",
+						"label": "Cluster extension resource name",
+						"defaultValue": "",
+						"toolTip": "Only lowercase alphanumeric characters are allowed, and the value must be 6-30 characters long.",
+						"constraints": {
+							"required": true,
+							"regex": "^[a-z0-9]{6,30}$",
+							"validationMessage": "Only lowercase alphanumeric characters are allowed, and the value must be 6-30 characters long."
+						},
+						"visible": true
+					},
+					{
+						"name": "title",
+						"type": "Microsoft.Common.TextBox",
+						"label": "Azure Vote Extension Title",
+						"toolTip": "This will replace the title's value inside your values.yaml",
+						"defaultValue": "Azure Vote App",
+						"constraints": {
+							"required": true,
+							"regex": "^[a-z0-9A-Z ]{1,30}$",
+							"validationMessage": "Only alphanumeric characters are allowed, and the value must be 6-30 characters long."
+						}
+					},
+					{
+						"name": "value1",
+						"type": "Microsoft.Common.TextBox",
+						"label": "Azure Vote Extension Option 1",
+						"toolTip": "This will replace the value1's value inside your values.yaml",
+						"defaultValue": "Cats",
+						"constraints": {
+							"required": true,
+							"regex": "^[a-z0-9A-Z]{1,30}$",
+							"validationMessage": "Only alphanumeric characters are allowed, and the value must be 6-30 characters long."
+						}
+					},
+					{
+						"name": "value2",
+						"type": "Microsoft.Common.TextBox",
+						"label": "Azure Vote Extension Option 2",
+						"toolTip": "This will replace the value2's value inside your values.yaml",
+						"defaultValue": "Dogs",
+						"constraints": {
+							"required": true,
+							"regex": "^[a-z0-9A-Z]{1,30}$",
+							"validationMessage": "Only alphanumeric characters are allowed, and the value must be 6-30 characters long."
+						}
+					},
+					{
+						"name": "autoUpgradeMinorVersion",
+						"type": "Microsoft.Common.CheckBox",
+						"label": "Autoupgrade Minor Version",
+						"toolTip": "Controls whether to allow auto upgrade of minor versions for the extension(application).",
+						"defaultValue": false
```

Comment:
should we set this to true by default?

Reply:
let me know if you see a benefit,  its a checkbox